request.go

```diff
@@ -38,11 +39,26 @@ type MessageImprint struct {
 }
 
 // Equal compares if m and n are the same MessageImprint
+//
+// Reference: RFC 3161 2.4.2
 func (m MessageImprint) Equal(n MessageImprint) bool {
 	return m.HashAlgorithm.Algorithm.Equal(n.HashAlgorithm.Algorithm) &&
+		m.HashAlgorithm.Parameters.Class == n.HashAlgorithm.Parameters.Class &&
+		m.HashAlgorithm.Parameters.Tag == n.HashAlgorithm.Parameters.Tag &&
+		m.HashAlgorithm.Parameters.IsCompound == n.HashAlgorithm.Parameters.IsCompound &&
+		bytes.Equal(m.HashAlgorithm.Parameters.Bytes, n.HashAlgorithm.Parameters.Bytes) &&
+		bytes.Equal(m.HashAlgorithm.Parameters.FullBytes, n.HashAlgorithm.Parameters.FullBytes) &&
 		bytes.Equal(m.HashedMessage, n.HashedMessage)
 }
 
+// ASN1NullRawValue represents the valid struct of asn1.NullRawValue
+//
+// https://pkg.go.dev/encoding/asn1#NullRawValue
+var ASN1NullRawValue = asn1.RawValue{
+	Tag:       5,
+	FullBytes: []byte{5, 0},
+}
```

Comment:
Why don't we use `asn1.NullRawValue` directly?

Reply:
This is explained in this PR's description: https://github.com/notaryproject/tspclient-go/pull/24#issue-2383266064